### PR TITLE
refactor trainer checks

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Part of the major test refactoring

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/base/eval_model_valid_epoch_ends.py [hunk 1]
```diff
@@ -34,8 +34,9 @@ def validation_epoch_end(self, outputs):
 
             val_acc_mean += val_acc
 
-        val_loss_mean /= len(outputs)
-        val_acc_mean /= len(outputs)
+        if outputs:  # skip zero divisions
```
Comment (tullie) [inline tests/base/eval_model_valid_epoch_ends.py:37]: Does this cover len(outputs) == 0 as well?

Comment (Borda) [reply tests/base/eval_model_valid_epoch_ends.py:37]: yes :]
```
output = []
bool(output)
Out[3]: False
```

File: tests/trainer/test_checks.py [hunk 2]
```diff
@@ -73,42 +55,28 @@ def test_warning_on_wrong_validation_settings(tmpdir):
...
     hparams = tutils.get_default_hparams()
-
-    trainer_options = dict(default_root_dir=tmpdir, max_epochs=1)
-    trainer = Trainer(**trainer_options)
-
-    class CurrentTestModel(LightTrainDataloader,
-                           LightValidationDataloader,
-                           TestModelBase):
-        pass
+    trainer = Trainer(efault_root_dir=tmpdir, max_epochs=1)
```
Comment (jeremyjordan) [inline tests/trainer/test_checks.py:58]: `efault_root_dir` typo?

Comment (Borda) [inline tests/trainer/test_checks.py:58]:
```suggestion
    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
```

File: tests/trainer/test_checks.py [hunk 3]
```diff
@@ -13,58 +13,40 @@
...
-            pass
-
-    trainer_options = dict(default_root_dir=tmpdir, max_epochs=1)
-    trainer = Trainer(**trainer_options)
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
 
     with pytest.raises(MisconfigurationException):
-        model = CurrentTestModel()
+        model = EvalModelTemplate(tutils.get_default_hparams())
+        model.configure_optimizers = LightningModule.configure_optimizers
```
Comment (williamFalcon) [inline tests/trainer/test_checks.py:45]: Why are you using LightningModule directly here?

Comment (williamFalcon) [reply tests/trainer/test_checks.py:45]: EvalModelTemplate has its own configure_optimizers

Comment (Borda) [reply tests/trainer/test_checks.py:45]: ok, I ll add empty config here

File: tests/base/eval_model_valid_epoch_ends.py [hunk 4]
```diff
@@ -17,8 +17,8 @@ def validation_epoch_end(self, outputs):
         # if returned a scalar from validation_step, outputs is a list of tensor scalars
         # we return just the average in this case (if we want)
         # return torch.stack(outputs).mean()
-        val_loss_mean = 0
-        val_acc_mean = 0
+        val_loss_mean = torch.tensor(0)
```
Comment (williamFalcon) [inline tests/base/eval_model_valid_epoch_ends.py:20]: this should be converted to the new syntax

```avg_val_loss = torch.stack(x['val_loss'] for x in outputs).mean()````
etc...

Comment (Borda) [inline tests/base/eval_model_valid_epoch_ends.py:20]:
```suggestion
        val_loss_mean = torch.stack(x['val_loss'] for x in outputs).mean()
```

File: tests/base/eval_model_valid_epoch_ends.py [hunk 5]
```diff
@@ -17,8 +17,8 @@ def validation_epoch_end(self, outputs):
...
+        val_acc_mean = torch.tensor(0)
```
Comment (Borda) [inline tests/base/eval_model_valid_epoch_ends.py:21]:
```suggestion
        val_acc_mean = torch.stack(x['val_acc'] for x in outputs).mean()
```

**Pull request merged**